doc/dev/cache-pool.rst

```diff
@@ -112,6 +113,17 @@ evict cache objects, all hit_set_count HitSets are loaded into RAM.
 Currently there is minimal benefit for hit_set_count > 1 since the
 agent does not yet act intelligently on that information.
 
+The ``min_read_recency_for_promote`` defines how many HitSets to check for the
+existence of an object when handling a read operation. The checking result is
+used to decide whether to promote the object asynchronously. Its value should be
+between 0 and ``hit_set_count``. If it's set to 0, the object is promoted anyway.
+If it's set to 1, the current HitSet is checked. And if this object is in the
+current HitSet, it's promoted. Otherwise not. For the other values, the exact
+number of archive HitSets are checked. The object is promoted if the object is
+found in any of the HitSets. ::
```

Comment:
@liewegas I think at first we just take the *recency* into account, as the name of this parameter stands for. Changing the behavior to be *all* instead of *any* sounds like a good idea, giving the fact that we know the less promotion, the better performance. And this takes the *frequency* into account as well.

How about this? We have `hit_set_count` hitsets in total. And we set `min_read_recency_for_promote` (or whatever names) to be a value between 0 and `hit_set_count`. We check object existence in all of the `hit_set_count` hitsets. When an object exists in no less than `min_read_recency_for_promote` hitsets, we promote it. This sounds more clear?

A drawback of this approach is that we need to maintain all of the `hit_set_count` hitsets in memory. Anyway, if we want to change the behavior of this parameter in either way, we'll need to do it in another PR. And change the name of this parameter accordingly?